tests/lib/rules/object-curly-newline.js

```diff
@@ -236,6 +236,111 @@ ruleTester.run("object-curly-newline", rule, {
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }]
+        },
+        {
+            code: [
+                "var c = {a: 1, b: 2};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }]
+        },
+        {
+            code: [
+                "var c = {",
+                "    a: 1,",
+                "    b: 2",
+                "};"
+            ].join("\n"),
+
+            options: [{ multiline: true, consistent: true }]
+        },
+        {
+            code: [
+                "var e = {a: function() { dosomething();}};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }]
+        },
+        {
+            code: [
+                "var e = {",
+                "    a: function() { dosomething();}",
+                "};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }]
+        },
+        {
+            code: [
+                "let {} = {a: 1};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: [
+                "let {a} = {a: 1};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: [
+                "let {",
+                "} = {a: 1};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: [
+                "let {",
+                "    a",
+                "} = {a: 1};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: [
+                "let {a, b} = {a: 1, b: 1};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: [
+                "let {",
+                "    a, b",
+                "} = {a: 1, b: 1};"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: [
+                "let {k = function() {dosomething();}} = obj;"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: [
+                "let {",
+                "    k = function() {",
+                "        dosomething();",
+                "    }",
+                "} = obj;"
+            ].join("\n"),
+            options: [{ multiline: true, consistent: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
```

Comment:
Could we add some tests for `{ multiline: false, consistent: true }` as well? For example, if I'm understanding the new option correctly, I would expect the following code to be valid when `consistent` is true but `multiline` is not:
```js
let {a,
b} = {a: 1, b: 2};
```